src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -667,15 +667,16 @@ bool tryGetToArrayMethod(TypeSymbol spreadTypeOriginalDefinition, WellKnownType
                 arrayType);
         }
 
-        /// <summary>
-        /// Returns true if type is convertible to Span or ReadOnlySpan.
-        /// If non-identity conversion, also returns a non-null asSpanMethod.
-        /// </summary>
+        /// <returns>
+        /// If <paramref name="writableOnly"/> is <see langword="true"/>, will only return true with a conversion to writable Span.
+        /// Otherwise, will return a conversion directly to ReadOnlySpan if one is known, and may return a conversion to Span if no direct conversion to ReadOnlySpan is known, or if the type is already a Span.
```

Comment:
>and may return a conversion to Span if no direct conversion to ReadOnlySpan is known, or if the type is already a Span. [](http://example.com/codeflow?start=90&length=119)

I think we should never be doing this for array types. In other words, if we do not intend to write into the span, we should never attempt to convert it to ```Span```

Reply:
It is quite possible that we are doing the right thing, but the comment is confusing. What does it mean for the conversion to be known?